web/src/cypress/support/alert.ts

```diff
@@ -246,7 +247,28 @@ function closeAlert(id: number): Cypress.Chainable<void> {
   return cy.graphqlVoid(query, { id })
 }
 
+function createAlertMetrics(alerts: Alert[]): Cypress.Chainable<void> {
+  const alertIDs: Array<number> = []
+  // build query
+  const query = `
+    insert into alert_metrics (alert_id, service_id, escalated, closed_at) 
+    select
+      a.id,
+      a.service_id,
+      (select count(*) > 1 from alert_logs where alert_id = a.id and event = 'escalated'),
+      (select timestamp from alert_logs where alert_id = a.id and event = 'closed' order by timestamp limit 1)
+    from alerts a
+    where a.id = any('{${alertIDs}}') and a.status='closed' and a.service_id is not null on conflict do nothing`
+
+  for (let i = 0; i < alerts.length; i++) {
+    alertIDs.push(Number(alerts[i].id))
+  }
+
+  return cy.sql(query)
+}
```

Comment:
Can we create & close alerts and rely on the engine to generate the metrics instead? The table schema can change over time, and the cypress tests should be user-interaction-level wherever possible.

Reply:
It can take up to two minutes for an alert to get pulled into the alert metrics table after it's closed, so this is trying to work around that time delay. We could add a delay to the cypress check instead.